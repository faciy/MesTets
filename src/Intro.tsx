import {View, Text} from 'react-native';
import React from 'react';

const Intro = props => {
  return (
    <View style={{flex: 1}} testID={'wrapper'}>
      <Text style={{color: 'red'}} testID={'text'}>
        {props.value}
      </Text>
    </View>
  );
};

export default Intro;
