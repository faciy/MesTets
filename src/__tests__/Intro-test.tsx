import React from 'react';
import renderer from 'react-test-renderer';
import Intro from '../Intro';
import {render, screen, fireEvent} from '@testing-library/react-native';

test('examples of some things', async () => {
  const expectedUsername = 'Ada Lovelace';
  render(<Intro />);

  fireEvent.changeText(screen.getByTestId('input'), expectedUsername);
  fireEvent.press(screen.getByText('Print Username'));

  const usernameOutput = await screen.findByTestId('printed-username');
  expect(usernameOutput).toHaveTextContent(expectedUsername);
  expect(screen.toJSON()).toMatchSnapshot();
});
