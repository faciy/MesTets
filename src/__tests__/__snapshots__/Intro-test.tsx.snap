// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rend correctement 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Bienvenue sur React Native !
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    C'est un test de snapshot pour React Native.
  </Text>
</View>
`;
